node-runner-cli/config/GatewayDockerConfig.py

```diff
@@ -36,117 +36,36 @@ class CoreApiNode(BaseConfig):
-                yield attr, dict(self.__getattribute__(attr))
-            elif self.__getattribute__(attr):
-                yield attr, self.__getattribute__(attr)
+class DataAggregatorConfig(BaseConfig):
+    release: str = None
+    repo: str = os.getenv(AGGREGATOR_DOCKER_REPO_OVERRIDE, "radixdlt/babylon-ng-data-aggregator")
+    restart: str = "unless-stopped"
+    NetworkName: str = None
+    coreApiNode: CoreApiNodeConfig = CoreApiNodeConfig({})
 
 
-class GatewayAPIDockerSettings(BaseConfig):
+class GatewayAPIDockerConfig(BaseConfig):
     release: str = None
-    repo: str = "radixdlt/babylon-ng-gateway-api"
-    coreApiNode: CoreApiNode = CoreApiNode({})
+    repo: str = os.getenv(GATEWAY_DOCKER_REPO_OVERRIDE, "radixdlt/babylon-ng-gateway-api")
+    coreApiNode: CoreApiNodeConfig = CoreApiNodeConfig({})
     restart = "unless-stopped"
     enable_swagger = "true"
     max_page_size = "30"
 
-    def ask_gateway_release(self):
-        latest_gateway_release = github.latest_release("radixdlt/babylon-gateway")
-        self.release = latest_gateway_release
-        if "DETAILED" in SetupMode.instance().mode:
-            self.release = Prompts.get_gateway_release("gateway_api", latest_gateway_release)
-
-    def set_core_api_node_setings(self, coreApiNode: CoreApiNode):
-        self.coreApiNode = coreApiNode
-
-    def __iter__(self):
-        class_variables = {key: value
-                           for key, value in self.__class__.__dict__.items()
-                           if not key.startswith('__') and not callable(value)}
-
-        for attr, value in class_variables.items():
-            if attr in ['coreApiNode']:
-                yield attr, dict(self.__getattribute__(attr))
-            elif self.__getattribute__(attr):
-                yield attr, self.__getattribute__(attr)
-
-
-class GatewayDockerSettings(BaseConfig):
-    data_aggregator = DataAggregatorSetting({})
-    gateway_api = GatewayAPIDockerSettings({})
-    postgres_db = PostGresSettings({})
-
-    def __iter__(self):
-        class_variables = {key: value
-                           for key, value in self.__class__.__dict__.items()
-                           if not key.startswith('__') and not callable(value)}
-
-        for attr, value in class_variables.items():
-            if attr in ['data_aggregator', 'gateway_api', 'postgres_db']:
-                yield attr, dict(self.__getattribute__(attr))
-            elif self.__getattribute__(attr):
-                yield attr, self.__getattribute__(attr)
 
-    def create_config(self, postgress_password):
-        self.data_aggregator.ask_core_api_node_settings()
-        self.postgres_db.ask_postgress_settings(postgress_password)
-        self.data_aggregator.ask_gateway_release()
-        self.gateway_api.ask_gateway_release()
-        self.gateway_api.set_core_api_node_setings(
-            self.data_aggregator.coreApiNode)
-        return self
+class GatewayDockerConfig(BaseConfig):
```

Comment:
And this is what I meant before. Here we have a config classes as elements of another config class when in reality they are all sort of siblings. What if we had 
GatewayDockerConfig(DataAggregatorConfigMixin, PostgresConfigMixin, DatabaseMigrationConfigMixin)